packages/wdio-browserstack-service/src/accessibility-handler.ts

```diff
@@ -0,0 +1,343 @@
+                            'filePath': this._suiteFile,
+                            'scopeList': [suiteTitle, test.title]
+                        },
+                        platform: this._platformA11yMeta
+                    }
+
+                    await this.sendTestStopEvent((this._browser as WebdriverIO.Browser), dataForExtension)
+
+                    if (shouldScanTestForAccessibility) {
+                        log.info('Accessibility testing for this test case has ended.')
+                    }
+                }
+            }
+        } catch (er) {
+            log.error(`Accessibility results could not be processed for the test case ${test.title}. Error :`, er)
+        }
+    }
+
+    /**
+      * Cucumber Only
+    */
+
+    async beforeScenario (world: ITestCaseHookParameter) {
+        if (!this._accessibility) {
+            return
+        }
+
+        const pickleData = world.pickle
+        const gherkinDocument = world.gherkinDocument
+        const featureData = gherkinDocument.feature
+        const uniqueId = getUniqueIdentifierForCucumber(world)
+
+        if (isBrowserstackSession(this._browser)) {
+            const shouldScanScenario = shouldScanTestForAccessibility(featureData?.name, pickleData.name, this._accessibilityOptions)
+            this._testMetadata[uniqueId] = {
+                scanTestForAccessibility : shouldScanScenario,
+                accessibilityScanStarted : true
+            }
+
+            if (this._browser && isAccessibilityAutomationSession(this._accessibility)) {
+                let pageOpen = true
+
+                try {
+                    const currentURL = await (this._browser as WebdriverIO.Browser).getUrl()
+                    const url = new URL(currentURL)
+                    pageOpen = url?.protocol === 'http:' || url?.protocol === 'https:'
+                } catch (e) {
+                    pageOpen = false
+                }
+
+                try {
+                    if (pageOpen) {
+                        if (shouldScanScenario) {
+                            log.info('Setup for Accessibility testing has started. Automate test case execution will begin momentarily.')
+                            await (this._browser as WebdriverIO.Browser).executeAsync(`
+                                const callback = arguments[arguments.length - 1]
+                                const fn = () => {
+                                    window.addEventListener('A11Y_TAP_STARTED', fn2)
+                                    const e = new CustomEvent('A11Y_FORCE_START')
+                                    window.dispatchEvent(e)
+                                }
+                                const fn2 = () => {
+                                    window.removeEventListener('A11Y_TAP_STARTED', fn)
+                                    callback()
+                                }
+                                fn()
```

Comment:
Is this the same function payload used as in `beforeTest` maybe worth putting it in an extra file, this makes it also easier to test this.